src/Carbon/CarbonInterval.php

```diff
@@ -1371,10 +1371,28 @@ public function forHumans($syntax = null, $short = false, $parts = -1, $options
             return $this->translate($unitData['unit'], $interpolations, $count, $translator, $altNumbers);
         };
 
-        foreach ($diffIntervalArray as $diffIntervalData) {
+        foreach ($diffIntervalArray as $index => $diffIntervalData) {
             if ($diffIntervalData['value'] > 0) {
                 $unit = $short ? $diffIntervalData['unitShort'] : $diffIntervalData['unit'];
                 $count = $diffIntervalData['value'];
+                if ($options & CarbonInterface::ROUND && count($interval) === $parts - 1 && $index < count($diffIntervalArray) - 1) {
+                    $roundValue = 0;
+                    $nextIntervalData = $diffIntervalArray[$index + 1];
+                    if ($nextIntervalData['unit'] === 'month') {
+                        $roundValue = $nextIntervalData['value'] / 12;
+                    } elseif ($nextIntervalData['unit'] === 'week') {
+                        $roundValue = $nextIntervalData['value'] / (365 / 12 / 7);
+                    } elseif ($nextIntervalData['unit'] === 'day') {
+                        $roundValue = $nextIntervalData['value'] / 7;
+                    } elseif ($nextIntervalData['unit'] === 'hour') {
+                        $roundValue = $nextIntervalData['value'] / 24;
+                    } elseif ($nextIntervalData['unit'] === 'minute') {
+                        $roundValue = $nextIntervalData['value'] / 60;
+                    } elseif ($nextIntervalData['unit'] === 'second') {
+                        $roundValue = $nextIntervalData['value'] / 60;
+                    }
```

Comment:
I guess we could have an array and only 1 if:
```php
$dividers = [
    'month' => 12,
    'week' => 365 / 12 / 7,
    'day' => 7,
    'hour' => 24,
    'minute' => 60,
    'second' => 60,
];

$divider = $dividers[$nextIntervalData['unit']] ?? null;

if ($divider) {
    $roundValue = $nextIntervalData['value'] / $divider;
}
```

Reply:
Moreover, we should use `static::getCascadeFactors()` so it takes the customization into account. and don't use magic numbers.
https://github.com/briannesbitt/Carbon/blob/master/src/Carbon/CarbonInterval.php#L236

This part is a bit tricky as cascade factors may skip a unit (for example day-to-month skipping week) so this enhancement may be for an other pull request.